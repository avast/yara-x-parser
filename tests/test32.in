//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		20% of ($a, $b)
}
