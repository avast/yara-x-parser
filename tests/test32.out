SOURCE_FILE@0..119
  RULE@0..118
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..118
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..84
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$a"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"foo\""
        WHITESPACE@71..74 "\n\t\t"
        VARIABLE_STMT@74..84
          VARIABLE@74..76 "$b"
          WHITESPACE@76..77 " "
          ASSIGN@77..78 "="
          WHITESPACE@78..79 " "
          PATTERN@79..84
            STRING_LIT@79..84 "\"bar\""
      WHITESPACE@84..86 "\n\t"
      CONDITION@86..116
        CONDITION_KW@86..95 "condition"
        COLON@95..96 ":"
        WHITESPACE@96..99 "\n\t\t"
        BOOLEAN_TERM@99..116
          OF_EXPR@99..116
            QUANTIFIER@99..104
              PRIMARY_EXPR@99..104
                L_PAREN@99..100 "("
                EXPRESSION@100..103
                  PRIMARY_EXPR@100..101
                    INT_LIT@100..101 "1"
                  PLUS@101..102 "+"
                  PRIMARY_EXPR@102..103
                    INT_LIT@102..103 "1"
                R_PAREN@103..104 ")"
            WHITESPACE@104..105 " "
            OF_KW@105..107 "of"
            WHITESPACE@107..108 " "
            PATTERN_IDENT_TUPLE@108..116
              L_PAREN@108..109 "("
              VARIABLE_WILDCARD@109..111
                VARIABLE@109..111 "$a"
              COMMA@111..112 ","
              WHITESPACE@112..113 " "
              VARIABLE_WILDCARD@113..115
                VARIABLE@113..115 "$b"
              R_PAREN@115..116 ")"
      WHITESPACE@116..117 "\n"
      R_BRACE@117..118 "}"
  WHITESPACE@118..119 "\n"
